packages/client/src/workflow-client.ts

```diff
@@ -640,6 +728,115 @@ export class WorkflowClient extends BaseClient {
     return await decodeFromPayloadsAtIndex(this.dataConverter, 0, response.queryResult?.payloads);
   }
 
+  /**
+   * Start Update and return a handle.
+   *
+   * Used as the final function of the interceptor chain during startUpdate and executeUpdate.
+   */
+  protected async _startUpdateHandler(
+    waitForStage: temporal.api.enums.v1.UpdateWorkflowExecutionLifecycleStage,
+    input: WorkflowStartUpdateInput
+  ): Promise<WorkflowStartUpdateOutput> {
+    const updateId = input.options?.updateId ?? uuid4();
+    const req: temporal.api.workflowservice.v1.IUpdateWorkflowExecutionRequest = {
+      namespace: this.options.namespace,
+      workflowExecution: input.workflowExecution,
+      firstExecutionRunId: input.firstExecutionRunId,
+      waitPolicy: { lifecycleStage: waitForStage },
+      request: {
+        meta: {
+          updateId,
+          identity: this.options.identity,
+        },
+        input: {
+          header: { fields: input.headers },
+          name: input.updateName,
+          args: { payloads: await encodeToPayloads(this.dataConverter, ...input.args) },
+        },
+      },
+    };
+    let response: temporal.api.workflowservice.v1.UpdateWorkflowExecutionResponse;
+
+    try {
+      response = await this.workflowService.updateWorkflowExecution(req);
+    } catch (err) {
+      this.rethrowGrpcError(err, 'Workflow Update failed', input.workflowExecution);
+    }
+    return {
+      updateId,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      workflowRunId: response.updateRef!.workflowExecution!.runId!,
+      outcome: response.outcome ?? undefined,
+    };
+  }
+
+  protected createWorkflowUpdateHandle<Ret>(
+    input: WorkflowStartUpdateInput,
+    output: WorkflowStartUpdateOutput
+  ): WorkflowUpdateHandle<Ret> {
+    const { updateId, workflowRunId } = output;
+    return {
+      updateId,
+      workflowId: input.workflowExecution.workflowId,
+      workflowRunId,
+      result: async () => {
+        const outcome = output.outcome ?? (await this._pollForUpdateOutcome(updateId, input.workflowExecution));
+        if (outcome.failure) {
+          throw new WorkflowUpdateFailedError(
+            'Workflow Update failed',
```

Comment:
Personally I am not convinced that lowercase "update" should be used to describe Temporal's Update operation (nor "query" nor "signal"). The phrase "update a workflow" might mean many things; for example it could mean "deploy new code fixing a bug in the workflow definition". So I am using uppercase Update everywhere. Let's discuss elsewhere and change across the board if we're changing.